### DISCUSSION TITLE
TeslaMate divides every drive into little parts

### DISCUSSION THREAD

**Discussion by rybber78** [teslamate-org/teslamate#1313] [Q&A]
**Describe the bug**

Every one of my drives are divided into small parts and counted as individual drives

**Expected behavior**

Should be listed as one drive. 

**How to reproduce it (as minimally and precisely as possible)**:

Go driving

(I'm running on docker and did not bother installing the MQTT part...)

**Relevant entries from the logs**

    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [is_preconditioning: false]]

14:22:17.846 [error] Task #PID<0.5210.0> started from #PID<0.5199.0> terminating

** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/update_available", "false", [retain: true, qos: 1]}, 10000)

    ** (EXIT) an exception was raised:

        ** (ErlangError) Erlang error: :timeout_value

            (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2

            (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4

            (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3

            (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4

            (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6

            (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3

    (elixir 1.10.4) lib/gen_server.ex:1023: GenServer.call/3

    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2

    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5

    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3

nction: &:erlang.apply/2

    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [update_available: false]]

      
      

```
paste logs here
```

**Screenshots**
![image](https://user-images.githubusercontent.com/41877177/97781628-08635b80-1b8d-11eb-82f4-e7c60282c650.png)

**Data**

**Environment**

- TeslaMate version: 1.20.1
- Type of installation: Docker
- OS TeslaMate is installed on: Linux
- User OS & Browser: Linux
- Others:

**Comment by ajeckmans** [comment]
I'm having the same issues. Once the car starts driving Teslamate starts receiving messages it chokes on. I haven't installed the MQTT  part either and have Teslamate running in docker.

@adriankumpf Can you tell me what info you still need?

**Relevant entries from the logs**
```
    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2,
    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
nction: &:erlang.apply/2,
    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [doors_open: false]],
14:32:22.794 [error] Task #PID<0.4387.0> started from #PID<0.4376.0> terminating,
** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/healthy", "true", [retain: true, qos: 1]}, 10000),
    ** (EXIT) an exception was raised:,
        ** (ErlangError) Erlang error: :timeout_value,
            (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2,
            (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4,
            (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3,
            (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4,
            (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6,
            (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
    (elixir 1.10.4) lib/gen_server.ex:1023: GenServer.call/3,
    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2,
    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
nction: &:erlang.apply/2,
    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [healthy: true]],
14:32:23.009 [error] GenServer TeslaMate.Mqtt.Publisher terminating,
** (stop) :timeout_value,
    (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2,
    (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4,
    (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3,
    (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4,
    (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
Last message (from #PID<0.4391.0>): {:publish, "teslamate/cars/1/display_name", "Auto", [retain: true, qos: 1]},
14:32:23.016 [error] Task #PID<0.4400.0> started from #PID<0.4389.0> terminating,
** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/healthy", "true", [retain: true, qos: 1]}, 10000),
    ** (EXIT) an exception was raised:,
        ** (ErlangError) Erlang error: :timeout_value,
            (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2,
            (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4,
            (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3,
            (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4,
            (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6,
            (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
    (elixir 1.10.4) lib/gen_server.ex:1023: GenServer.call/3,
    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2,
    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
nction: &:erlang.apply/2,
    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [healthy: true]],
14:32:23.016 [error] Task #PID<0.4391.0> started from #PID<0.4389.0> terminating,
** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/display_name", "Auto", [retain: true, qos: 1]}, 10000),
    ** (EXIT) an exception was raised:,
        ** (ErlangError) Erlang error: :timeout_value,
            (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2,
            (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4,
            (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3,
            (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4,
            (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6,
            (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
    (elixir 1.10.4) lib/gen_server.ex:1023: GenServer.call/3,
    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2,
    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
nction: &:erlang.apply/2,
    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [display_name: "Auto"]],
14:32:23.023 [error] Task #PID<0.4397.0> started from #PID<0.4389.0> terminating,
** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/longitude", "5.97926", [retain: true, qos: 1]}, 10000),
    ** (EXIT) an exception was raised:,
        ** (ErlangError) Erlang error: :timeout_value,
            (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2,
            (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4,
            (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3,
            (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4,
            (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6,
            (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
    (elixir 1.10.4) lib/gen_server.ex:1023: GenServer.call/3,
    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2,
    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
nction: &:erlang.apply/2,
    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [longitude: 5.97926]],
14:32:23.024 [error] Task #PID<0.4396.0> started from #PID<0.4389.0> terminating,
** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/inside_temp", "21.5", [retain: true, qos: 1]}, 10000),
    ** (EXIT) an exception was raised:,
        ** (ErlangError) Erlang error: :timeout_value,
            (tortoise 0.9.5) lib/tortoise/connection.ex:328: Tortoise.Connection.connection/2,
            (tortoise 0.9.5) lib/tortoise.ex:269: Tortoise.publish/4,
            (teslamate 1.20.1) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3,
            (stdlib 3.12.1) gen_server.erl:661: :gen_server.try_handle_call/4,
            (stdlib 3.12.1) gen_server.erl:690: :gen_server.handle_msg/6,
            (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
    (elixir 1.10.4) lib/gen_server.ex:1023: GenServer.call/3,
    (elixir 1.10.4) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2,
    (elixir 1.10.4) lib/task/supervised.ex:35: Task.Supervised.reply/5,
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3,
nction: &:erlang.apply/2,
    Args: [#Function<3.4852099/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [inside_temp: 21.5]],
14:32:23.078 [info] Application teslamate exited: shutdown,
{"Kernel pid terminated",application_controller,"{application_terminated,teslamate,shutdown}"}
,
Kernel pid terminated (application_controller) ({application_terminated,teslamate,shutdown})
,

,
Crash dump is being written to: erl_crash.dump...done
,
14:32:40.194 [info] Migrations already up,
```

**Environment**
-  TeslaMate version: 1.20.1
-  Type of installation: Docker
-  OS TeslaMate is installed on: Linux
-  User OS & Browser: Windows / Firefox
-  Others:

**Comment by ajeckmans** [comment]
Never mind... I forgot to set DISABLE_MQTT in the env variables. @rybber78 did you perhaps forget as well?

**Comment by rybber78** [comment]
Yup, can confirm DISABLE_MQTT=true setting fixes it. Please close.

**Comment by mischu81** [comment]
Where did you enter this variable?

**Comment by brendanpward** [comment]
When starting the docker container, it's set as an environment variable with the -e switch.
`docker run -e DISABLE_MQTT=true teslamate/teslamate:latest`

**Comment by mischu81** [comment]
> When starting the docker container, it's set as an environment variable with the -e switch.
> `docker run -e DISABLE_MQTT=true teslamate/teslamate:latest`

Super! it finally works. Thank you

**Comment by Strobi-Boy** [comment]
Hello,
since the last Update (1.20 -> 1.20.1), I have the same problem with the devided drives.
    TeslaMate version: 1.20.1
    Type of installation: Docker
    OS TeslaMate is installed on: RaspBerry Stretch
    User OS & Browser: Windows / Firefox
    
![grafik](https://user-images.githubusercontent.com/74811819/99882633-47cc1780-2c22-11eb-8baf-318afbcda955.png)

I've tried the workaround
**docker run -e DISABLE_MQTT=true teslamate/teslamate:latest** 
but it didn't work with my system.
![grafik](https://user-images.githubusercontent.com/74811819/99882708-ca54d700-2c22-11eb-9d78-0d0db1a7fad6.png)

Do you have some other ideas to fix the problem?

Thank you very much.

**Comment by brendanpward** [comment]
You'll need to include a few more environment variables than just MQTT_DISABLE such as DATABASE_USER, DATABASE_PASS, etc. As seen at https://docs.teslamate.org/docs/configuration/environment_variables/.  I just shortened my example down to the one variable for brevity. Each one will require its own -e switch.

docker run -e MQTT_DISABLE=true -e DATABASE_USER=the_user -e DATABASE_PASS=mypassword -e DATABASE_HOST=dbserver.org -e DATABASE_PORT=5432 -e DATABASE_NAME=mydb teslamate/teslamate:latest

There are a 

If you're using the Docker compose method seen here https://docs.teslamate.org/docs/installation/docker then you'll want to add MQTT_DISABLE=true under the environment list, while removing MQTT_HOST=mosquitto. 


-------- Original Message --------

**Comment by Strobi-Boy** [comment]
Thanks for the fast response.
I have replaced "MQTT_HOST=mosquitto" with " MQTT_DISABLE=true" and started the docker again "docker -compose up -d" - but it seems, that I have also other troubles?

![grafik](https://user-images.githubusercontent.com/74811819/99884011-3cc9b500-2c2b-11eb-8506-4bfecd858580.png)

What is going wrong with my TeslaMate?
I also have installed the openWB at this RaspBerry.

Thanks for your advice.

**Comment by JonMurphy** [comment]
I'm experiencing the exact same problem as Strobi-Boy with `waiting for Postgres at 127.0.0.1... ` And I'm also using the Docker version.  I am guessing I'm not stopping, pausing or "down"ing Docker in the right way.

My guess is the steps should be:
* Stop Docker
* Change docker-compose.yml
* Start Docker

But I am too new at Docker to figure out what I am doing wrong.  (I'm still learning about Docker but I haven't come across the right ways to do this!)

**Comment by ajeckmans** [comment]
You don't need to stop docker. Do a "docker-compose up" again after changing the file. It will update the configuration and restart any containers it needs to.

**Comment by Strobi-Boy** [comment]
Hello,
I have changed the "compose.yml"...
![grafik](https://user-images.githubusercontent.com/74811819/99944687-9a691900-2d73-11eb-980f-d297d78e2b1c.png)

...and restartet the Docker - but the problem with the splitted drives is already there:
![grafik](https://user-images.githubusercontent.com/74811819/99944828-d7cda680-2d73-11eb-8f3d-8eb814cceffc.png)

Do you have any more tips to solve the problem?
Thank you.

**Comment by Strobi-Boy** [comment]
Do anyone has any idea, how to fix the problem?
It was since the last TeslaMate - Update...

**Comment by ajeckmans** [comment]
You'll need to add some logs from your container as well as the entire compose file and your currently running docker containers (docker ps) . There's not enough information right now to help.

If I had to guess one of the components is not running properly after you tried to install it twice.

**Comment by Strobi-Boy** [comment]
Hello,
where do I find the logs from my container?

**This is the compose file:**
version: "3"



services:

  teslamate:

    image: teslamate/teslamate:latest

    restart: always

    environment:

      - DATABASE_USER=teslamate

      - DATABASE_PASS=secret

      - DATABASE_NAME=teslamate

      - DATABASE_HOST=database

      - MQTT_DISABLE=true

    ports:

      - 4000:4000

    volumes:

      - ./import:/opt/app/import

    cap_drop:

      - all



  database:

    image: postgres:13

    restart: always

    environment:

      - POSTGRES_USER=teslamate

      - POSTGRES_PASSWORD=secret

      - POSTGRES_DB=teslamate

    volumes:

      - teslamate-db:/var/lib/postgresql/data


grafana:

    image: teslamate/grafana:latest

    restart: always

    environment:

      - DATABASE_USER=teslamate

      - DATABASE_PASS=secret

      - DATABASE_NAME=teslamate

      - DATABASE_HOST=database

    ports:

      - 3000:3000

    volumes:

      - teslamate-grafana-data:/var/lib/grafana

 mosquitto:

    image: eclipse-mosquitto:1.6

    restart: always

    ports:

      - 1883:1883

    volumes:

      - mosquitto-conf:/mosquitto/config

      - mosquitto-data:/mosquitto/data



volumes:

  teslamate-db:

  teslamate-grafana-data:

  mosquitto-conf:

  mosquitto-data:



**This are the runnig docker containers:**

pi@RaspBerryPi:~ $ docker ps
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS                    NAMES
8e97b234e274        postgres:13                  "docker-entrypoint.s…"   3 days ago          Up 3 days           5432/tcp                 pi_database_1
bb29a3dd2403        teslamate/grafana:latest     "/run.sh"                7 days ago          Up 3 days           0.0.0.0:3000->3000/tcp   default_grafana_1
ce006396bdf6        teslamate/teslamate:latest   "/sbin/tini -- /bin/…"   7 days ago          Up 4 hours          0.0.0.0:4000->4000/tcp   default_teslamate_1
2e99bd1e16ac        postgres:12                  "docker-entrypoint.s…"   7 days ago          Up 3 days           5432/tcp                 default_database_1

**Comment by DrMichael** [comment]
docker-comopose logs
oder
docker-comopose logs -f

Oder
docker-comopose logs teslamate

**Comment by Strobi-Boy** [comment]
Here we go:

**$ docker-compose logs**
/usr/local/lib/python3.5/dist-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release ofcryptography. Please upgrade your Python.
  from cryptography.hazmat.backends import default_backend
Attaching to pi_mosquitto_1, pi_grafana_1, pi_teslamate_1, pi_database_1
database_1   | 
database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2020-11-21 17:47:13.459 UTC [1] LOG:  starting PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on arm-unknown-linux-gnueabihf, compiled by gcc (Debian 8.3.0-6) 8.3.0, 32-bit
database_1   | 2020-11-21 17:47:13.459 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2020-11-21 17:47:13.459 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1   | 2020-11-21 17:47:13.474 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1   | 2020-11-21 17:47:13.510 UTC [27] LOG:  database system was shut down at 2020-11-21 17:46:56 UTC
database_1   | 2020-11-21 17:47:13.560 UTC [1] LOG:  database system is ready to accept connections
database_1   | 2020-11-21 17:47:42.103 UTC [1] LOG:  received fast shutdown request
database_1   | 2020-11-21 17:47:42.404 UTC [1] LOG:  aborting any active transactions
database_1   | 2020-11-21 17:47:42.411 UTC [1] LOG:  background worker "logical replication launcher" (PID 33) exited with exit code 1
database_1   | 2020-11-21 17:47:42.415 UTC [28] LOG:  shutting down
database_1   | 2020-11-21 17:47:44.697 UTC [1] LOG:  database system is shut down
database_1   | 
database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2020-11-21 17:49:50.717 UTC [1] LOG:  starting PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on arm-unknown-linux-gnueabihf, compiled by gcc (Debian 8.3.0-6) 8.3.0, 32-bit
database_1   | 2020-11-21 17:49:50.719 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2020-11-21 17:49:50.719 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1   | 2020-11-21 17:49:50.740 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1   | 2020-11-21 17:49:50.779 UTC [26] LOG:  database system was shut down at 2020-11-21 17:47:44 UTC
database_1   | 2020-11-21 17:49:50.938 UTC [1] LOG:  database system is ready to accept connections


**$ docker-compose logs -f**
/usr/local/lib/python3.5/dist-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release ofcryptography. Please upgrade your Python.
  from cryptography.hazmat.backends import default_backend
Attaching to pi_mosquitto_1, pi_grafana_1, pi_teslamate_1, pi_database_1
database_1   | 
database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2020-11-21 17:47:13.459 UTC [1] LOG:  starting PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on arm-unknown-linux-gnueabihf, compiled by gcc (Debian 8.3.0-6) 8.3.0, 32-bit
database_1   | 2020-11-21 17:47:13.459 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2020-11-21 17:47:13.459 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1   | 2020-11-21 17:47:13.474 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1   | 2020-11-21 17:47:13.510 UTC [27] LOG:  database system was shut down at 2020-11-21 17:46:56 UTC
database_1   | 2020-11-21 17:47:13.560 UTC [1] LOG:  database system is ready to accept connections
database_1   | 2020-11-21 17:47:42.103 UTC [1] LOG:  received fast shutdown request
database_1   | 2020-11-21 17:47:42.404 UTC [1] LOG:  aborting any active transactions
database_1   | 2020-11-21 17:47:42.411 UTC [1] LOG:  background worker "logical replication launcher" (PID 33) exited with exit code 1
database_1   | 2020-11-21 17:47:42.415 UTC [28] LOG:  shutting down
database_1   | 2020-11-21 17:47:44.697 UTC [1] LOG:  database system is shut down
database_1   | 
database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2020-11-21 17:49:50.717 UTC [1] LOG:  starting PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on arm-unknown-linux-gnueabihf, compiled by gcc (Debian 8.3.0-6) 8.3.0, 32-bit
database_1   | 2020-11-21 17:49:50.719 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2020-11-21 17:49:50.719 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1   | 2020-11-21 17:49:50.740 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1   | 2020-11-21 17:49:50.779 UTC [26] LOG:  database system was shut down at 2020-11-21 17:47:44 UTC
database_1   | 2020-11-21 17:49:50.938 UTC [1] LOG:  database system is ready to accept connections
pi_mosquitto_1 exited with code 128
pi_teslamate_1 exited with code 128
pi_grafana_1 exited with code 128

**$ docker-compose logs teslamate**
/usr/local/lib/python3.5/dist-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release ofcryptography. Please upgrade your Python.
  from cryptography.hazmat.backends import default_backend
Attaching to pi_teslamate_1

**$ docker-compose up -d**
/usr/local/lib/python3.5/dist-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release ofcryptography. Please upgrade your Python.
  from cryptography.hazmat.backends import default_backend
Starting pi_teslamate_1 ... 
pi_database_1 is up-to-date
Starting pi_teslamate_1 ... error
Starting pi_mosquitto_1 ... 
WARNING: Host is already in use by another container

**ERROR**: for pi_teslamate_1  Cannot start service teslamate: driver failed programming external connectivity on endpoint pi_teslamate_1 (b96220ed3dd4368e93b52139e65a3144e0672414742b1922d10b2e0174f126d5): BiStarting pi_grafana_1   ... error
WARNING: Host is already in use by another container

**ERROR**: for pi_grafana_1  Cannot start service grafana: driver failed programming external connectivity on endpoint pi_grafana_1 (447f47e6bd3183c5f3589d08f686155fc05b3690fdd27435d3bfe774f3899f92): Bind for 0.0.0.0:3000 failed: port is already allocated
Starting pi_mosquitto_1 ... error

**ERROR**: for pi_mosquitto_1  Cannot start service mosquitto: driver failed programming external connectivity on endpoint pi_mosquitto_1 (58423f4d4025742c9450ee87337984236f02b4dec7d4b2c17ecc694eb5f19a35): Error starting userland proxy: listen tcp 0.0.0.0:1883: bind: address already in use

**ERROR**: for grafana  Cannot start service grafana: driver failed programming external connectivity on endpoint pi_grafana_1 (447f47e6bd3183c5f3589d08f686155fc05b3690fdd27435d3bfe774f3899f92): Bind for 0.0.0.0:3000 failed: port is already allocated

**ERROR**: for mosquitto  Cannot start service mosquitto: driver failed programming external connectivity on endpoint pi_mosquitto_1 (58423f4d4025742c9450ee87337984236f02b4dec7d4b2c17ecc694eb5f19a35): Error starting userland proxy: listen tcp 0.0.0.0:1883: bind: address already in use

**ERROR**: for teslamate  Cannot start service teslamate: driver failed programming external connectivity on endpoint pi_teslamate_1 (b96220ed3dd4368e93b52139e65a3144e0672414742b1922d10b2e0174f126d5): Bind for 0.0.0.0:4000 failed: port is already allocated
ERROR: Encountered errors while bringing up the project.

**Comment by Strobi-Boy** [comment]
No idea why drives are splitted?
Charges will work correctly!

**Comment by Strobi-Boy** [comment]
If there is no possibility to fix the problem, is it possible save the TeslaMate Data and to remove the TeslaMate / Docker installation completely?
How to restore the TeslaMate data after the reinstall?

**Comment by Strobi-Boy** [comment]
Hello,
I have reinstalled my Raspberry with Stretch & openWB - I have the same problems with Teslamate.
All drives are devides into small parts.

![grafik](https://user-images.githubusercontent.com/74811819/101350924-1d1fc700-3890-11eb-92dc-3eee90d9f07f.png)
![grafik](https://user-images.githubusercontent.com/74811819/101351059-4b9da200-3890-11eb-8f11-5d01bf6f0790.png)

The following logs are attached:
[Teslamate Log.txt](https://github.com/adriankumpf/teslamate/files/5652670/Teslamate.Log.txt)
[Teslamge Logs.txt](https://github.com/adriankumpf/teslamate/files/5652672/Teslamge.Logs.txt)

**Comment by Strobi-Boy** [comment]
> docker ps

pi@raspberrypi:~ $ docker ps
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS                          PORTS                    NAMES
7c1467b5341b        teslamate/teslamate:latest   "/sbin/tini -- /bin/…"   2 minutes ago       Restarting (1) 28 seconds ago                            pi_teslamate_1
fdaaa48487e5        postgres:13                  "docker-entrypoint.s…"   3 days ago          Up 3 days                       5432/tcp                 pi_database_1
f509f707f65c        teslamate/grafana:latest     "/run.sh"                9 days ago          Up 8 days                       0.0.0.0:3000->3000/tcp   pi_grafana_1

![grafik](https://user-images.githubusercontent.com/74811819/102339725-e172b480-3f95-11eb-81fd-d145b3f9ded2.png)

**Comment by Strobi-Boy** [comment]
I've tried a little bit and changed the mosquitto port from 1883 to 1882 and now it seems, that it work - but not 100% correct:
![grafik](https://user-images.githubusercontent.com/74811819/102682678-b1145b80-41cb-11eb-8f8a-d2fdc9946578.png)

The old data ist gone - but the drives will be displayed correct - but without informations about the consumption:
![grafik](https://user-images.githubusercontent.com/74811819/102682718-f9cc1480-41cb-11eb-8521-4dada19c257e.png)
 
Some informations in the drive details are also missing:
![grafik](https://user-images.githubusercontent.com/74811819/102682760-46afeb00-41cc-11eb-8023-9f2d40d6ac93.png)

**Comment by ajeckmans** [comment]
@Strobi-Boy your fist issue was caused by you accidentally installing teslamate twice. The remnants of the fist installation were reserving ports which caused the second installation to fail to start properly. 

Your second issue seems to be caused by something holding on to the default port of mosquitto, Did you completely wipe your raspberry pi or are there still some docker images running from the previous installs?

You then changed the port mapping of your MQTT container, so any connection made to 1883 from the outside is going to go to 1882. This is fine as long as you instruct the MQTT process inside the container to also bind to 1882 which I don't see you doing. Resulting in any data sent by Teslamate to mosquitto:1883 to end up in a black hole. I'm actually surprised it's not causing a fatal error in Teslamate.

**Comment by Strobi-Boy** [comment]
In the meantime I've tried several new installations (both with stretch and buster).
- New installation:
sudo apt-get update && sudo apt-get upgrade && sudo apt-get dist-upgrade
- sudo reboot
curl -sSL https://get.docker.com | sh
sudo usermod -aG docker pi
- sudo reboot
docker run hello-world
sudo apt-get install -y libffi-dev libssl-dev
sudo apt-get install -y python3 python3-pip
sudo apt-get remove python-configparser
sudo apt install postgresql libpq-dev postgresql-client 
sudo apt-get update && sudo apt-get upgrade
sudo pip3 -v install docker-compose
curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && sudo python3 get-pip.py
sudo apt-get update && sudo apt-get install
sudo apt-get update && sudo apt-get dist-upgrade
- sudo reboot
sudo nano docker-compose.yml (insert Script from https://docs.teslamate.org/docs/installation/docker/)
docker-compose up -d

Every time I first install TeslaMate and then OpenWB (to control the wallbox), TeslaMate splits the trips into small parts.
How can I check, which tools will connect to the default port of mosquitto? Because I have made some new & clean installs and it will take place every time. Teslamate is the only image which will run in docker.

How can I change the default-port of 1883 to another port (including MQTT process), so that Teslamate can work as it shoud?

**Comment by ajeckmans** [comment]
> In the meantime I've tried several new installations (both with stretch and buster).

Are you sure any traces of the old installations you tried are gone?

> How can I check, which tools will connect to the default port of mosquitto? 

`lsof -i | grep LISTEN` will probably work for this

**Comment by Strobi-Boy** [comment]
> Are you sure any traces of the old installations you tried are gone?

Yes - I installed a new and fresh Buster / Stretch (FULL) with the few steps I wrote.
So it was a clean installation with no old connections.

I changed the port so 1883:1833 and started Teslamate again:
![grafik](https://user-images.githubusercontent.com/74811819/103096222-854a0900-4603-11eb-8b15-6dfb6b4c1223.png)

> `lsof -i | grep LISTEN` will probably work for this

It will show this:
![grafik](https://user-images.githubusercontent.com/74811819/103096278-b4f91100-4603-11eb-8345-78f701be4f8f.png)

To install OpenWB I only ran the following commands:
`sudo apt-get install jq`
`curl -s https://raw.githubusercontent.com/snaptec/openWB/master/openwb-install.sh | sudo sh`
Then I edited two OpenWB files in the text editor.
And that's it - unfortunately Teslamate no longer works.
